src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -142,13 +142,13 @@ static int32_t guest_surface_path_finding(Peep* peep)
     int16_t z = peep->next_z;
```

Comment:
Wouldn't it be better to just create a CoordsXYRangedZ and feed that into the functions? That should make the code easier to read.

Alternatively, making `z` use big coordinates would also clean up a lot already.